catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/SubjectCommands.java

```diff
@@ -33,72 +35,93 @@
             "-u"}, multiValued = false, description = "Run command as a different user")
     protected String user = null;
 
+    protected SubjectCommands() {
+        this(Security.getInstance());
+    }
+
+    SubjectCommands(Security security) {
+        this.security = security;
+    }
+
+    /**
+     * Executes the command once the user has been properly authorized.
+     *
+     * @return result of the command execution
+     * @throws Exception if the command failed to run
+     */
     protected abstract Object executeWithSubject() throws Exception;
 
     /**
-     * doExecute of SubjectCommands attemps to run a command as a certain subject.
-     * The order it checks for subject information is
-     * 1. User supplied subject name via command line
-     * 2. Shiro thread context subject
-     * 3. Java Subject
-     * If there is a java subject and it has a RolePrincipal for admin the system subject is used.
-     * If no valid subject is found an error message will be printed to the console and the
-     * command will not be executed. Even if a valid subject is found, that subject might not have
-     * the permissions nessessary to run the command.
+     * Executes the command using the user name provided using the {@code --user} option and
+     * prompts for a password. If no user name was provided, tries to run the command using the
+     * current {@link Subject}, or elevates to the system subject is the user has permission to do
+     * so.
+     * <p/>
+     * An error message will be printed out if the user name/password combination is invalid or
+     * if the user doesn't have the permission to run the command.
      *
-     * @return
-     * @throws Exception
+     * @return value returned by {@link #executeWithSubject()}, or {@code null} if the command failed
+     * @throws InvocationTargetException thrown if an exception occurred while executing
+     *                                   the command
+     * @throws Exception                 if any other unexpected exception occurred
      */
     @Override
     protected Object doExecute() throws Exception {
-        Subject subject = null;
-        if (!StringUtils.isEmpty(user)) {
-            String password = getLine("Password for " + user + ": ", false);
-            subject = Security.getSubject(user, password);
-        } else {
-            try {
-                //check for a shiro subject
-                subject = org.apache.shiro.SecurityUtils.getSubject();
-            } catch (Exception e) {
-                LOGGER.debug("No shiro subject available for running command");
-            }
 
-            if (subject == null) {
-                //verify that java subject has the correct roles (admin)
-                if (!Security.javaSubjectHasAdminRole()) {
-                    printErrorMessage(
-                            "Current user doesn't have sufficient privileges to run this command");
+        try {
+            if (!StringUtils.isEmpty(user)) {
```

Comment:
There is a an isNotBlank function: https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#isNotBlank(java.lang.CharSequence)